### PR TITLE
Login handler

### LINKED ISSUE #86: [Backend | Login & Auth API] Login Handler

### PR DESCRIPTION
**Pull request by Jacob4649**
Closes #86.

### CODE REVIEW

File: model/src/main/java/com/javahelp/model/util/json/ProviderUserInfoConverter.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * @return {@link ProviderUserInfoConverter} instance
+     */
+    public static ProviderUserInfoConverter getInstance() {
+        return instance;
+    }
+
+    @Override
```
Comment (Lam207) [inline model/src/main/java/com/javahelp/model/util/json/ProviderUserInfoConverter.java:31]: Seems good to me, I would change it later if this is merged before `Gender` PR.

File: model/src/main/java/com/javahelp/model/util/json/ClientUserInfoConverter.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    /**
+     * @return {@link ClientUserInfoConverter} instance
+     */
+    public static ClientUserInfoConverter getInstance() {
+        return instance;
+    }
+
+    @Override
```
Comment (Lam207) [inline model/src/main/java/com/javahelp/model/util/json/ClientUserInfoConverter.java:31]: This is also good, but I will update `Gender` attribute when this PR is merged first before the `Gender` PR

**Review by Lam207 — APPROVED** [review]
Great way of implementation, especially for handling exceptions!

File: backend/src/test/java/com/javahelp/backend/domain/user/login/LoginInteractorTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        ClientUserInfo info = new ClientUserInfo("sdjkfhkls", "sdf", "sdf",
+                "sdfhjkl", "sdjlfhkslkjd");
+
+        User u = new User("", info, "testing_user_login_123");
+
+        db.create(u, p);
+
+        LoginInteractor interactor = new LoginInteractor(db, tokens);
+
+        LoginResult result = interactor.login(new ILoginInput() {
```
Comment (stevolopolis) [inline backend/src/test/java/com/javahelp/backend/domain/user/login/LoginInteractorTest.java:35]: I'm not too familiar, but is this necessarily a good practice? To implement an interface in a test class?

Comment (Jacob4649) [reply backend/src/test/java/com/javahelp/backend/domain/user/login/LoginInteractorTest.java:35]: Yeah using an anonymous implementation keeps everything simple!

**Review by stevolopolis — APPROVED** [review]
Besides the implementation of interface in LoginInteractorTest which I am unsure off, I think this PR is good to go.
Will make some changes to the UserInfoConvertors in my next PR to incorporate the surveyResponses.

**Comment by Jacob4649** [comment]
> Besides the implementation of interface in LoginInteractorTest which I am unsure of (resolved), I think this PR is good to go. Will make some changes to the UserInfoConvertors in my next PR to incorporate the surveyResponses.

Awesome sounds good!

**Review by Enkhzul001 — APPROVED** [review]
Everything looks great

**Pull request merged**